scripts/parse_new_build_warnings.py

```diff
@@ -91,21 +94,28 @@ def RemoveFromSet(msg: str):
     RemoveFromSet(parsed)
     return new_build_warnings
 
-def append_warnings_to_file(output_file: str, warning_set: Set[str], old_build_name: str, new_build_name: str):
-    """Append the warnings to a given file_path"""
-    output_path = Path(output_file)
-    # Create possible missing output file directory
-    output_path.parent.mkdir(parents=True, exist_ok=True)
-
-    with open(output_path, 'a') as file:
-        introduction = f"# New Warnings from new build: {new_build_name} old build: {old_build_name}\n\n"
-        file.write(introduction)
-        for warning in warning_set:
-            file.write(warning)
-        # Add extra new lines for better visual
-        file.write("\n\n")
-
-def parse_new_build_warnings_from_directory(old_build_directory: str, new_build_directory: str, output: str):
+def parse_target(file_name: str)-> str:
+    """Parse only the target info from the post commit build log formatted file name"""
+    parts = file_name.split('-')
+    # Replace the hash with a wildcard regex
+    HASH_INDEX = 3
+    del parts[HASH_INDEX]
+    BUILD_LOG_INDEX=-3
+    return '-'.join(parts[:BUILD_LOG_INDEX])
+
+def convert_pre_to_post_format(file_name: str):
+    """Convert pre-commit's build log file format to post-commit's format
+    pre-commit's build log format: patch_number-mode-arch-abi-multilib-build-log-stderr.log
+    post-commit's build log format: mode-arch-abi-hash-multilib-build-log-stderr.log
+    """
+    parts = file_name.split('-')
+    PATCH_NAME_INDEX = 0
+    HASH_INDEX = 3
+    patch_name = parts.pop(PATCH_NAME_INDEX)
+    parts.insert(HASH_INDEX, patch_name)
```

Comment:
why is the patch number the hash?

Reply:
The patch number is the unique attribute for the pre-commit format that can act as post-commit's hash. Since pre-commit's build log format doesn't involve any build hash, I needed a unique identifier

Reply:
The patch number is the unique attribute for the pre-commit format that can act as post-commit's hash. Since pre-commit's build log format doesn't involve any build hash, I needed a unique identifier